sim/hunter/mongoose_bite.go

```diff
@@ -0,0 +1,110 @@
+	"github.com/wowsims/sod/sim/core/proto"
+	"github.com/wowsims/sod/sim/core/stats"
+)
+
+func (hunter *Hunter) getMongooseBiteConfig(rank int) core.SpellConfig {
+	spellId := [5]int32{0, 1495, 14269, 14270, 14271}[rank]
+	baseDamage := [5]float64{0, 25, 45, 75, 115}[rank]
+	manaCost := [5]float64{0, 30, 40, 50, 65}[rank]
+	level := [5]int{0, 16, 30, 44, 58}[rank]
+
+	raptorFuryDmgMult := 0.1
+	spellConfig := core.SpellConfig{
+		ActionID:      core.ActionID{SpellID: spellId},
+		SpellSchool:   core.SpellSchoolPhysical,
+		DefenseType:   core.DefenseTypeMelee,
+		ProcMask:      core.ProcMaskMeleeMHSpecial,
+		Flags:         core.SpellFlagMeleeMetrics | core.SpellFlagAPL,
+		Rank:          rank,
+		RequiredLevel: level,
+
+		ManaCost: core.ManaCostOptions{
+			FlatCost: manaCost,
+		},
+
+		Cast: core.CastConfig{
+			DefaultCast: core.Cast{
+				GCD: core.GCDDefault,
+			},
+			CD: core.Cooldown{
+				Timer:    hunter.NewTimer(),
+				Duration: time.Second * 5,
+			},
+		},
+
+		ExtraCastCondition: func(sim *core.Simulation, target *core.Unit) bool {
+			return hunter.DistanceFromTarget <= 5 && hunter.DefensiveState.IsActive()
+		},
+
+		BonusCritRating:  float64(hunter.Talents.SavageStrikes) * 10 * core.CritRatingPerCritChance,
+		CritDamageBonus: hunter.mortalShots(),
+		DamageMultiplier: 1,
+
+		ApplyEffects: func(sim *core.Simulation, target *core.Unit, spell *core.Spell) {
+			hunter.DefensiveState.Deactivate(sim)
+
+			multiplier := 1.0
+			if stacks := hunter.RaptorFuryAura.GetStacks(); stacks > 0 {
+				multiplier *= 1 + raptorFuryDmgMult*float64(stacks)
+			}
+			
+			damage := multiplier * (baseDamage + (hunter.GetStat(stats.AttackPower) * 0.4))
+			spell.CalcAndDealDamage(sim, target, damage, spell.OutcomeMeleeWeaponSpecialHitAndCrit)
+		},
+	}
+
+	return spellConfig
+}
+
+func (hunter *Hunter) registerCobraSlayerAura() {
+	hunter.RegisterAura(core.Aura{
+		Label:    "Cobra Slayer Trigger",
+		Duration: core.NeverExpires,
+		MaxStacks: 20,
+		OnReset: func(aura *core.Aura, sim *core.Simulation) {
+			aura.Activate(sim)
+		},
+		OnSpellHitDealt: func(aura *core.Aura, sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
+			if result.Outcome.Matches(core.OutcomeDodge) {
+				aura.SetStacks(sim, 1);
+				hunter.DefensiveState.Activate(sim)
+			} else if result.Outcome.Matches(core.OutcomeLanded) {
+				if spell.ProcMask == core.ProcMaskMeleeMHAuto || spell.ProcMask == core.ProcMaskMeleeOHAuto {
+					if sim.Proc((float64(aura.GetStacks()) * 0.05), "Cobra Slayer") {
+						aura.SetStacks(sim, 1);
+						hunter.DefensiveState.Activate(sim)
+					} else {
+						aura.AddStack(sim)
+					}
+				}
+			}
+		},
+	})
+
+	hunter.DefensiveState = hunter.RegisterAura(core.Aura{
+		Label:    "Defensive State",
+		ActionID: core.ActionID{SpellID: 5302},
+		Duration: time.Second * 5,
+	})
+}
+
+func (hunter *Hunter) registerMongooseBiteSpell() {
+	if !hunter.HasRune(proto.HunterRune_RuneChestCobraSlayer) {
+		return
+	}
```

Comment:
Since Mongoose bite exists as a separate ability from the Cobra Slayer rune, could we separate the creation of the cobra slayer aura from registering mongoose bite? It could get in `hunter/runes.go` instead

Reply:
Hunter tank dream prayge, but I agree it's probably better to implement the spell correctly rather than quickly!